torch/_inductor/codegen/cuda/gemm_template.py

```diff
@@ -0,0 +1,472 @@
+
+        # Skip simt kernels
+        if (
+            op.tile_description.math_instruction.opcode_class
+            == cutlass_lib.OpcodeClass.Simt
+        ):
+            return None
+
+        # Skip sparse kernels
+        if op.gemm_kind == cutlass_lib.GemmKind.Sparse:
+            return None
+
+        # Filter ops by dtypes.
+        X = self.input_nodes[0]
+        W = self.input_nodes[1]
+        accumulator_torch_dtype = cutlass_utils.get_accumulator_dtype(
+            [X.get_dtype(), W.get_dtype()],
+        )
+        if not (
+            cutlass_utils.dtype_match(X.get_dtype(), op.A.element)
+            and cutlass_utils.dtype_match(W.get_dtype(), op.B.element)
+            and cutlass_utils.dtype_match(self.output_node.get_layout().dtype, op.C.element)
+            and cutlass_utils.dtype_match(accumulator_torch_dtype, op.accumulator_type())
+        ):
+            return None
+
+        # Filter ops by input layouts.
+        if not (
+            self.layout_match(X.get_layout(), op.A.layout)
+            and self.layout_match(W.get_layout(), op.B.layout)
+        ):
+            return None
+
+        # Update op.
+        op = copy.deepcopy(op)
+
+        # Set output layout.
+        op.D.layout = CutlassGemmTemplate.cutlass_layout(self.output_node.get_layout())
+
+        # Filter ops by alignments and set alignments.
+        if not (
+            self.set_alignment(X.get_layout(), op.A)
+            and self.set_alignment(W.get_layout(), op.B)
+            and self.set_alignment(self.output_node.get_layout(), op.D)
+        ):
+            return None
+
+        # Set epilogue.
+        # TODO: update epilogue functor according to epilogues.
+        op.element_epilogue = op.accumulator_type()
+
+        # Set bias layout and alignment.
+        if len(self.input_nodes) >= 3 and self.input_nodes[2] is not None:
+            Bias = self.input_nodes[2]
+            op.C.layout = CutlassGemmTemplate.cutlass_layout(Bias.get_layout())
+            if not self.set_alignment(Bias.get_layout(), op.C):
+                return None
+        else:
+            if op.gemm_kind == cutlass_lib.GemmKind.Universal3x:
+                op.C.element = cutlass_lib.DataType.void
+            else:
+                op.C.layout = op.D.layout
+
+        return op
+
+    def gen_ops(self) -> List[cutlass_gemm_op.GemmOperation]:
+        ops = cutlass_utils.gen_ops()[cutlass_lib.OperationKind.Gemm]
+        res = set()
+        num_3x_ops = 0
+        num_2x_ops = 0
+        for key, op_list in ops.items():
+            for op in op_list:
+                filter_res = self.filter_op(op)
+                if filter_res is not None:
+                    res.add(filter_res)
+        for op in res:
+            if op.gemm_kind == cutlass_lib.GemmKind.Universal3x:
+                num_3x_ops += 1
+            else:
+                num_2x_ops += 1
+        print(f"Got cutlass configs: {len(res)=}, {num_3x_ops=}, {num_2x_ops=}")
```

Comment:
Leftover print statement from development?

Reply:
Yes didn't fix it in draft code ^^.